### DISCUSSION TITLE
Custom Images Not Displaying on Deployed CKAN Instance

### DISCUSSION THREAD

**Discussion by Perla-Al-Haddad** [ckan/ckan#6909] [Q&A]
**CKAN version**
2.9.5
Deployed CKAN docker instance on Linux VM

**Describe the bug**
When deploying a local version of CKAN to a Linux VM with Ubuntu 18.04, the images in the cutsom extension created throw a 404 error in the browser and don't display. They display normally when running the instance locally.

**Steps to reproduce**
1. Setup ckan using the package installation on WSL 2 on windows 10 with Ubuntu 20.04. 
2. Create a CKAN extension and follow the documentation [here](https://docs.ckan.org/en/2.9/theming/static-files.html) to add images.
3. Deploy the CKAN instance on a Linux VM using the docker-compose installation.

**Expected behavior**
The images added should show up on the deployed instance the same way they do on the local instance.

**Additional details**
Image error log:
![SS ckan 404 image](https://user-images.githubusercontent.com/89902296/173435106-3056f928-ee80-413b-ac08-1c193d972e5c.PNG)
![SS ckan 404 image 2](https://user-images.githubusercontent.com/89902296/173435702-a45471b2-4671-4983-bdd4-a39037412b8e.PNG)

Extension Plugin Code Snippet:
![SS ckan code snippet](https://user-images.githubusercontent.com/89902296/173437416-60461f2d-6d6c-446b-9c9a-31c1409276dd.PNG)

**Comment by amercader** [comment]
@Perla-Al-Haddad It looks like the image is being referenced from a CSS directive. Does the image path there look fine? What is the full URL that is generated and returns the 404.

Can you also check that the image file is stored in 

```
ckanext-datasettheme/
  ckanext/
    datasettheme/
      public/
        images/
          headers/
            visualization.jpg
            
```

**Reply by Perla-Al-Haddad** [reply]
@amercader 
The full URL that is generating the 404 error is:
```
http://<ipaddress>:<portnumber>/images/headers/visualization.jpg
```
And yes, the image is stored in:
```
ckanext_datasettheme/ckanext/datasettheme/public/images/headers/visualization.jpg
```
I'm not sure I understand what you mean by the image is being referenced from a CSS directive. But this is how the image is being referenced in the css: 
```css
url("/images/headers/visualization.jpg")
```

Also, a temporary solution I found was moving the images to the `ckan/public/base/images` folder, and removing this line from the plugin.py file for the extension:
```python
toolkit.add_public_directory(config_, 'public');
```
But I know that this is not the proper way of doing things, and I'd rather find a way to fix this that keeps my extension files seperate from the source ckan files.

Thank you for your help! 🙏

**Comment by mareksloser** [comment]
Hello all, @amercader 


I have same problem - i have no folder after extension install.

But before installation all assets are in folder.

"docker-ckan\images\ckan\2.9\ext\ckanext\NAME_theme\public\assets"

![image](https://user-images.githubusercontent.com/10614710/187911226-1c31f605-df44-4d80-992e-135753601db2.png)


![image](https://user-images.githubusercontent.com/10614710/187910765-dda6dfd7-202e-4867-8963-e1124f30049d.png)

**Comment by vanquan223** [comment]
I added `*.jpg *.png` to recursive-include in MANIFEST.in file and it solved the problem
`recursive-include ckanext/xxx*.html *.json *.js *.less *.css *.mo *.jpg *.png`
